plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -384,24 +416,85 @@ public void setViewPrincipal(ConnectorSession session, SchemaTableName schemaVie
+        Map<SchemaTableName, ConnectorViewDefinition> views = new HashMap<>();
+        for (Namespace restNamespace : namespaces) {
+            List<TableIdentifier> restViews = restSessionCatalog.listViews(sessionContext, restNamespace);
+            for (TableIdentifier restView : restViews) {
+                SchemaTableName schemaTableName = SchemaTableName.schemaTableName(restView.namespace().toString(), restView.name());
+                getView(session, schemaTableName).ifPresent(view -> views.put(schemaTableName, view));
+            }
+        }
+
+        return views;
     }
 
     @Override
     public Optional<ConnectorViewDefinition> getView(ConnectorSession session, SchemaTableName viewName)
     {
+        try {
+            View view = restSessionCatalog.loadView(convert(session), toIdentifier(viewName));
+            Optional<String> comment = Optional.ofNullable(view.properties().get(COMMENT));
+            ViewVersion currentVersion = view.currentVersion();
+            Optional<ConnectorViewDefinition> viewDefinition = Optional.empty();
+            List<ConnectorViewDefinition.ViewColumn> viewColumns = IcebergUtil.viewColumnsFromSchema(typeManager, view.schema());
+
+            for (ViewRepresentation representation : currentVersion.representations()) {
+                if (representation instanceof SQLViewRepresentation) {
+                    SQLViewRepresentation sqlView = (SQLViewRepresentation) representation;
+                    String sql = sqlView.sql();
+                    Optional<String> defaultCatalog = Optional.ofNullable(currentVersion.defaultCatalog());
+                    Optional<String> defaultNamespace = Optional.of(currentVersion.defaultNamespace().toString());
+                    if (defaultCatalog.isEmpty()) {
+                        defaultCatalog = Optional.of(catalogName.toString());
+                    }
+
+                    // Found a trino dialect view, so that should get used
+                    if (sqlView.dialect().equals(TRINO_SQL_DIALECT)) {
+                        Optional<String> owner = Optional.ofNullable(view.properties().get(TRINO_OWNER));
+                        boolean runAsInvoker = Boolean.parseBoolean(view.properties().get(TRINO_RUN_AS_INVOKER));
+                        ConnectorViewDefinition trinoViewDefinition = new ConnectorViewDefinition(
+                                sql, defaultCatalog, defaultNamespace, viewColumns, comment, owner, runAsInvoker, null);
+                        viewDefinition = Optional.of(trinoViewDefinition);
+                        break;
+                    }
+
+                    // Set a best effort view definition
+                    viewDefinition = Optional.of(new ConnectorViewDefinition(
+                            sql, defaultCatalog, defaultNamespace, viewColumns, comment, Optional.empty(), false, null));
+                }
+
+                return viewDefinition;
+            }
```

Comment:
When view is found but doesn't have a working SQLViewRepresentation, we should throw instead of returning `Optional.empty`. This ensures consistency between listing views and getting views -- anything that shows up on the listing cannot later be considered "not found".

Reply:
Definitely want consistency between listing and getting views, but what's the rationale for throwing instead of returning Optional.empty() in both? Update so that it's consistent between both but we return Optional.empty

Reply:
@findepi The behavior is consistent but it's not throwing an exception, and rather returning Optional.empty. let me know if you see any issues with that.